text/0000-export-to-oci.md

```diff
@@ -0,0 +1,691 @@
+                └── oci-layout
+
+```
+
+As we can see, the application image `my-app-image` has missing `blobs` because they were not provided as input and the lifecycle just **skip writing** those layers on disk.
+
+##### Using -layout flag in combination with --daemon or --publish flags
+
+Any combination of using multiple sources or sinks in the Lifecycle invocation of phases should throw an error to the user. For example:
+
+```=shell
+> export CNB_USE_OCI_LAYOUT=true
+> /cnb/lifecycle/exporter -daemon -run-image cnb/my-full-stack-run:bionic my-app-image
+
+# OR
+
+> /cnb/lifecycle/exporter -layout -daemon -run-image cnb/my-full-stack-run:bionic my-app-image
+
+ERROR: exporting to multiples target is not allowed
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The lifecycle phases affected by this new behavior are:
+ - [Analyze](https://buildpacks.io/docs/concepts/components/lifecycle/analyze/)
+ - [Restore](https://buildpacks.io/docs/concepts/components/lifecycle/restore/)
+ - [Export](https://buildpacks.io/docs/concepts/components/lifecycle/export/)  
+ - [Create](https://buildpacks.io/docs/concepts/components/lifecycle/create/)
+
+At a high level view the proposed solution can be summarized with the following system landscape diagram from the C4 model
+
+![](https://i.imgur.com/y972lTD.png)
+
+Notice that we are relying on the OCI format Specification to expose the data for `Platforms`
+
+The following new inputs are proposed to be added to these phases
+
+ | Input | Environment Variable  | Default Value | Description
+ |-------|-----------------------|---------------|--------------
+ | `<layout>` | `CNB_USE_OCI_LAYOUT` | false | Enables the capability of resolving image from/to in OCI layout format on disk |
+
+## How to map an image reference into a path in the layout repository
+
+In the previous examples one key element was how to translate an image reference into a path, let's define those rules.
+
+Considering an **image reference** refers to either a tag reference or digest reference. It could have the following formats
+- A name reference refers to an identifier of form `<registry>/<repo>/<image>:<tag>`
+- A digest reference refers to a content addressable identifier of form `<registry>/<repo>/<image>@<algorithm>:<digest>`
+
+The image look up will be done following these rules:
+  - WHEN `the image points to a name reference`
+    - Lifecycle will load/save the image from/to disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `/<registry>/<repo>/<image>/<tag>`
+  - WHEN `the image points to a digest reference`
+    - Lifecycle will load the image from disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `/<registry>/<repo>/<image>/<algorithm>/<digest>`
+  - WHEN `<registry>` is not provided default value will be **index.docker.io**
+    - IF `<repo>` is not also provided, then default value will be **library**
```

Comment:
Aren't we always expecting the lifecycle to be provided a path? And from there we can derive the name if needed?

Because otherwise we are always expecting the images to be in the `/` directory, which is doubtful.

Reply:
`<layout-dir>` was added again, without default value, now platforms MUST set this value and avoid mounting into `/` directory.